---
title: "Maize harvest"
author: "Jan WÃ¤lchli, Selma Cadot and Klaus Schlaeppi"
output:
  pdf_document: 
    toc: yes
    toc_depth: 3
---
   
```{r setup, include=FALSE, , echo=F}

#clear the object from memory
rm(list=ls())

#knitr settings
knitr::opts_chunk$set(echo = TRUE, fig.align = "center")

#set seed
set.seed(100)

```

```{r libraries, echo=F, message=F, warning=F}

#set source to file location
if (!require(rstudioapi)){install.packages("rstudioapi")}; library(rstudioapi)
setwd(dirname(getActiveDocumentContext()$path))

#paths
paths <- list(interim = c("../1_data/interim"))

#load functions
source("functions/libraries.R")
source("functions/variability_table.R")

#installs (if necessary) and loads libraries
libraries()

#ggplot settings
theme_set(theme_bw(base_size = 12))

```

```{r paths, echo=F, message=F, warning=F}

#set wd to RDS files
setwd(paths$interim)

#import RDS files
bDESIGN <- readRDS("bDESIGN.RDS")
fDESIGN <- readRDS("fDESIGN.RDS")

bDAT <- readRDS("bDAT.RDS")
fDAT <- readRDS("fDAT.RDS")
b_rare <- readRDS("b_rare.RDS")
f_rare <- readRDS("f_rare.RDS")
bDAT_rare <- readRDS("bDAT_rare.RDS")
fDAT_rare <- readRDS("fDAT_rare.RDS")

bTAX <- readRDS("bTAX.RDS")
fTAX <- readRDS("fTAX.RDS")

level_cols_Sample_type <- readRDS("level_cols_Sample_type.RDS")
level_cols_Conditioning <- readRDS("level_cols_Conditioning.RDS")

```

```{r subsetting to MAIZE samples, echo=F, warning=F, message=F, eval=T, }

#Bacteria
#subset DESIGN
bDESIGN_maize <- droplevels(bDESIGN[bDESIGN$Species=="maize", ])
rownames(bDESIGN_maize) <- bDESIGN_maize$bprimer
bDESIGN_maize_root <- droplevels(bDESIGN_maize[bDESIGN_maize$Sample_type == "root",])
bDESIGN_maize_rhizo <- droplevels(bDESIGN_maize[bDESIGN_maize$Sample_type == "rhizosphere",])
bDESIGN_maize_soil <- droplevels(bDESIGN_maize[bDESIGN_maize$Sample_type == "soil",])

#subset DATA_rare
bDAT_rare_maize <- bDAT_rare[ ,rownames(bDESIGN_maize)]
bDAT_rare_maize <- bDAT_rare_maize[rowSums(bDAT_rare_maize) > 0,]  # removal of rows with 0 values
bDAT_rare_maize_root <- bDAT_rare_maize[ ,rownames(bDESIGN_maize_root)]
bDAT_rare_maize_root <- bDAT_rare_maize_root[rowSums(bDAT_rare_maize_root) > 0,]  # removal of rows with 0 values
bDAT_rare_maize_rhizo <- bDAT_rare_maize[ ,rownames(bDESIGN_maize_rhizo)]
bDAT_rare_maize_rhizo <- bDAT_rare_maize_rhizo[rowSums(bDAT_rare_maize_rhizo) > 0,]  # removal of rows with 0 values
bDAT_rare_maize_soil <- bDAT_rare_maize[ ,rownames(bDESIGN_maize_soil)]
bDAT_rare_maize_soil <- bDAT_rare_maize_soil[rowSums(bDAT_rare_maize_soil) > 0,]  # removal of rows with 0 values

#make PHYLOSEQ
bPHYSEQ_rare_maize <- phyloseq(sample_data(bDESIGN_maize),
                               otu_table(bDAT_rare_maize, taxa_are_rows=T),
                               tax_table(as.matrix(bTAX[rownames(bDAT_rare_maize),])))

#Fungi
#subset DESIGN
fDESIGN_maize <- droplevels(fDESIGN[fDESIGN$Species=="maize", ])
rownames(fDESIGN_maize) <- fDESIGN_maize$fprimer
fDESIGN_maize_root <- droplevels(fDESIGN_maize[fDESIGN_maize$Sample_type == "root",])
fDESIGN_maize_rhizo <- droplevels(fDESIGN_maize[fDESIGN_maize$Sample_type == "rhizosphere",])
fDESIGN_maize_soil <- droplevels(fDESIGN_maize[fDESIGN_maize$Sample_type == "soil",])

#subset DATA_rare
fDAT_rare_maize <- fDAT_rare[ ,rownames(fDESIGN_maize)]
fDAT_rare_maize <- fDAT_rare_maize[rowSums(fDAT_rare_maize) > 0,]  # removal of rows with 0 values
fDAT_rare_maize_root <- fDAT_rare_maize[ ,rownames(fDESIGN_maize_root)]
fDAT_rare_maize_root <- fDAT_rare_maize_root[rowSums(fDAT_rare_maize_root) > 0,]  # removal of rows with 0 values
fDAT_rare_maize_rhizo <- fDAT_rare_maize[ ,rownames(fDESIGN_maize_rhizo)]
fDAT_rare_maize_rhizo <- fDAT_rare_maize_rhizo[rowSums(fDAT_rare_maize_rhizo) > 0,]  # removal of rows with 0 values
fDAT_rare_maize_soil <- fDAT_rare_maize[ ,rownames(fDESIGN_maize_soil)]
fDAT_rare_maize_soil <- fDAT_rare_maize_soil[rowSums(fDAT_rare_maize_soil) > 0,]  # removal of rows with 0 values

#make PHYLOSEQ
fPHYSEQ_rare_maize <- phyloseq(sample_data(fDESIGN_maize),
                               otu_table(fDAT_rare_maize, taxa_are_rows=T),
                               tax_table(as.matrix(fTAX[rownames(fDAT_rare_maize),])))

```

\pagebreak 

# Data analysis - Maize harvest

Here we analysed the microbiome data which has been collected during maize harvesting.

# Soil-chemical gradient

We know from previous analyses that we have a soil-chemical gradient along the field. We tested if the microbiome changes along this gradient. We tested this effect with a PERMANOVA and illustrated it with a PCoA.

## PERMANOVA

*Model: ~ Sample_type + Soil Chemistry PC1*

```{r bray curtis distances, warning=F, echo=F, eval=T}

#Bacteria
bDAT_rare_maize_dist <- vegdist(t(bDAT_rare_maize), method="bray")
bDAT_rare_maize_root_dist <- vegdist(t(bDAT_rare_maize_root), method="bray")
bDAT_rare_maize_rhizo_dist <- vegdist(t(bDAT_rare_maize_rhizo), method="bray")
bDAT_rare_maize_soil_dist <- vegdist(t(bDAT_rare_maize_soil), method="bray")

#bDAT_rare_maize_dist_paov <- adonis2(bDAT_rare_maize_dist ~ Sample_type + pos_long + pos_lat, data=bDESIGN_maize, permutations=999)
bDAT_rare_maize_dist_paov <- adonis2(bDAT_rare_maize_dist ~ Sample_type + PCA_soil_chem_1, data=bDESIGN_maize, permutations=999)
rownames(bDAT_rare_maize_dist_paov)[1] <- "Sampletype"
rownames(bDAT_rare_maize_dist_paov)[2] <- "Soil chemistry PC1"  
pander(bDAT_rare_maize_dist_paov[1:3,c(3,5)], caption="Bacteria")

#Fungi
fDAT_rare_maize_dist <- vegdist(t(fDAT_rare_maize), method="bray")
fDAT_rare_maize_root_dist <- vegdist(t(fDAT_rare_maize_root), method="bray")
fDAT_rare_maize_rhizo_dist <- vegdist(t(fDAT_rare_maize_rhizo), method="bray")
fDAT_rare_maize_soil_dist <- vegdist(t(fDAT_rare_maize_soil), method="bray")

#fDAT_rare_maize_dist_paov <- adonis2(fDAT_rare_maize_dist ~ Sample_type + pos_long + pos_lat, data=fDESIGN_maize, permutations=999)
fDAT_rare_maize_dist_paov <- adonis2(fDAT_rare_maize_dist ~ Sample_type + PCA_soil_chem_1, data=fDESIGN_maize, permutations=999)
rownames(fDAT_rare_maize_dist_paov)[1] <- "Sampletype"
rownames(fDAT_rare_maize_dist_paov)[2] <- "Soil chemistry PC1"  
pander(fDAT_rare_maize_dist_paov[1:3,c(3,5)], caption="Fungi")

```

\pagebreak

## PCoA

We performed an unconstrained ordination with Bray-Curtis distances.

### Figure 3 | PCoA with Bray-Curtis 

```{r bacteria PCoA all samples, echo=F, warning=F, message=F, eval=T, fig.height=6, fig.width=11}

#Bacteria
#PCoA plot
bPHYSEQ_rare_maize_PCoA_bray <- phyloseq::ordinate(bPHYSEQ_rare_maize, method="PCoA", distance="bray")
p0 <- phyloseq::plot_ordination(bPHYSEQ_rare_maize, bPHYSEQ_rare_maize_PCoA_bray, shape="Sample_type", color="PCA_soil_chem_1", title = "Bacteria at maize harvest (PCoA, Bray-Curtis)", axes = c(1,2))
p0 <- p0 + geom_point(size=3)
p0 <- p0 + scale_shape_manual(values=c(16,17,15))
p0 <- p0 +  theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
                  legend.margin = margin(b =  0, unit='cm'),
                  legend.box.margin = margin(b = -10, unit = "pt"))
Fig_2B_bac <- p0


#Fungi
#PCoA plot
fPHYSEQ_rare_maize_PCoA_bray <- phyloseq::ordinate(fPHYSEQ_rare_maize, method="PCoA", distance="bray")
p0 <- phyloseq::plot_ordination(fPHYSEQ_rare_maize, fPHYSEQ_rare_maize_PCoA_bray, shape="Sample_type", color="PCA_soil_chem_1", title = "Fungi at maize harvest (PCoA, Bray-Curtis)", axes = c(1,2))
p0 <- p0 + geom_point(size=3)
p0 <- p0 + scale_shape_manual(values=c(16,17,15)) 
p0 <- p0 +  theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
                  legend.margin = margin(b =  0, unit='cm'),
                  legend.box.margin = margin(b = -10, unit = "pt"))
Fig_2B_fun <- p0

#combine plots (package:cowplot)
legend <- get_legend(Fig_2B_bac + theme(legend.position="bottom") + labs(shape="Sampletype", col="Soil Chemistry PC1"))
p <- plot_grid(Fig_2B_bac + theme(legend.position="none"),
               Fig_2B_fun + theme(legend.position="none"),
                  labels=c(" "," "), align = "hv", nrow=1)

p <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(p)

```

\pagebreak

## Correlation analysis: PCoA ~ position 

Do the field positions correlate with the beta diversity? We tested for the first two axes.

### Figure 4.1 | Bacteria: Correlation PCoA ~ position

```{r correlation bacteria, warning=F, echo=F, eval=T, message=F, fig.height=3, fig.width=11}

#function to convert r to R2
r_to_R2 <- function(r){paste0((r^2)*100,"%")}

#function plot pos long
plot_cor <- function(df, cor, PCoA_axis, ylabel, pos_field){
  plot <- ggplot(df, aes(y=get(PCoA_axis), x=get(pos_field), colour = Sample_type)) +
            geom_jitter(width = 0.1, size=3, aes(shape=Sample_type)) + 
            scale_colour_manual(values = df$cols_Sample_type) +
            stat_smooth(method = "lm", se = FALSE, fullrange = TRUE, colour = "dimgrey", alpha = 3/4) +
            scale_x_continuous(breaks=seq(1, max(df[,paste(pos_field)]), 1), limits=c(1, max(df[,paste(pos_field)]))) +
            theme_bw(base_size = 12) +
            theme(legend.position="none") +
            xlab("Field position: length") +
            ylab(ylabel) +
            labs(title = paste("R2:", r_to_R2(round(cor$estimate,2)), "p:", formatC(cor$p.value, format = "g", digits = 2)))
  return(plot)
}

#function plot pos wide
plot_cor2 <- function(df, cor, PCoA_axis, ylabel, pos_field){
  plot <- ggplot(df, aes(y=get(PCoA_axis), x=get(pos_field), colour = Sample_type)) +
            geom_jitter(width = 0.1, size=3, aes(shape=Sample_type)) + 
            scale_colour_manual(values = df$cols_Sample_type) +
            stat_smooth(method = "lm", se = FALSE, fullrange = TRUE, colour = "dimgrey", alpha = 3/4) +
            scale_x_continuous(breaks=seq(1, max(df[,paste(pos_field)]), 1), limits=c(1, max(df[,paste(pos_field)]))) +
            theme_bw(base_size = 12) +
            theme(legend.position="none") +
            xlab("Field position: width") +
            ylab(ylabel) +
            labs(title = paste("R2:", r_to_R2(round(cor$estimate,2)), "p:", formatC(cor$p.value, format = "g", digits = 2)))
  return(plot)
}

#get PCoA dataframe
bPHYSEQ_PCoA_bray_df <- phyloseq::plot_ordination(bPHYSEQ_rare_maize, bPHYSEQ_rare_maize_PCoA_bray, justDF = T)

#correlation tests (PCoA ~ position)
bPHYSEQ_PCoA_bray_df_A1_long <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.1, bPHYSEQ_PCoA_bray_df$pos_long)
bPHYSEQ_PCoA_bray_df_A2_long <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.2, bPHYSEQ_PCoA_bray_df$pos_long)
bPHYSEQ_PCoA_bray_df_A1_lat <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.1, bPHYSEQ_PCoA_bray_df$pos_lat)
bPHYSEQ_PCoA_bray_df_A2_lat <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.2, bPHYSEQ_PCoA_bray_df$pos_lat)

#make plots
p1 <- plot_cor(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A1_long, "Axis.1", "PCoA Axis 1", "pos_long")
p2 <- plot_cor(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A2_long, "Axis.2", "PCoA Axis 2", "pos_long")
p3 <- plot_cor2(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A1_lat, "Axis.1", "PCoA Axis 1", "pos_lat")
p4 <- plot_cor2(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A2_lat, "Axis.2", "PCoA Axis 2", "pos_lat")
  
#combine plots
legend <- get_legend(p1 + theme(legend.position="bottom") + labs(shape="Sampletype", col="Sampletype"))
p <- plot_grid(p1 + theme(legend.position="none"),
               p2 + theme(legend.position="none"),
               p3 + theme(legend.position="none"),
               p4 + theme(legend.position="none"),
               labels=c(" "," "), align = "hv", nrow=1)

bCORplot <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(bCORplot)

```

### Figure 4.2 | Fungi: Correlation PCoA ~ position

```{r correlation Fungi, warning=F, echo=F, eval=T, message=F, fig.height=3, fig.width=11}

#get PCoA dataframe
fPHYSEQ_PCoA_bray_df <- phyloseq::plot_ordination(fPHYSEQ_rare_maize, fPHYSEQ_rare_maize_PCoA_bray, justDF = T)

#correlation tests (PCoA ~ position)
fPHYSEQ_PCoA_bray_df_A1_long <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.1, fPHYSEQ_PCoA_bray_df$pos_long)
fPHYSEQ_PCoA_bray_df_A2_long <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.2, fPHYSEQ_PCoA_bray_df$pos_long)
fPHYSEQ_PCoA_bray_df_A1_lat <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.1, fPHYSEQ_PCoA_bray_df$pos_lat)
fPHYSEQ_PCoA_bray_df_A2_lat <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.2, fPHYSEQ_PCoA_bray_df$pos_lat)

#make plots
p1 <- plot_cor(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A1_long, "Axis.1", "PCoA Axis 1", "pos_long")
p2 <- plot_cor(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A2_long, "Axis.2", "PCoA Axis 2", "pos_long")
p3 <- plot_cor2(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A1_lat, "Axis.1", "PCoA Axis 1", "pos_lat")
p4 <- plot_cor2(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A2_lat, "Axis.2", "PCoA Axis 2", "pos_lat")
  
#combine plots
legend <- get_legend(p1 + theme(legend.position="bottom") + labs(shape="Sampletype", col="Sampletype"))
p <- plot_grid(p1 + theme(legend.position="none"),
               p2 + theme(legend.position="none"),
               p3 + theme(legend.position="none"),
               p4 + theme(legend.position="none"),
               labels=c(" "," "), align = "hv", nrow=1)

fCORplot <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(fCORplot)

```

*Conclusion:* The microbiome is influenced by the chemical soil gradient. To take this chemical soil gradient into account, we included the axis 1 of the PCA of the soil chemistry data as independent variable in all following models.

\pagebreak

# Analysis of BX conditioning effects - Alpha diversity

How did BX conditioning affect the microbiome? We compared the root, rhizosphere and soil microbiomes of maize whether the microbial communities differ between the conditioning treatments (Bx+, Bx-). To measure the alpha diversity, we rarefied the data to the sequencing depth of `r b_rare` (bacteria) and `r f_rare` (fungi). Then we calculated Shannon diversity in each sample. This was repeated 100 times and the mean value from the 100 iterations was taken for statistical analysis between the different samples.  

```{r a-div shannon, warning=F, echo=F, message=F, eval=T}

#function to calculate Shannon
get_shannon <- function(DAT, rare){
  #get shannon diversity
  diversity_all_DAT_mat <- c()
  for (i in 1:100){
    #generate rarefied community
    temp_mat <- t(rrarefy(t(DAT), rare))
    shannon <- vegan::diversity(temp_mat, index="shannon", MARGIN=2)
    d <- exp(shannon)
    diversity_all_DAT_mat <- cbind(diversity_all_DAT_mat, d)
    }
    return(diversity_all_DAT_mat)
}

#shannon div
bDAT_temp <- bDAT[, rownames(bDESIGN_maize)]
bDAT_temp <- bDAT_temp[rowSums(bDAT_temp) > 0,]  #removal of rows with 0 values
bALPHA_mat <- get_shannon(bDAT_temp, b_rare)
rm(bDAT_temp)

fDAT_temp <- fDAT[, rownames(fDESIGN_maize)]
fDAT_temp <- fDAT_temp[rowSums(fDAT_temp) > 0,]  # emoval of rows with 0 values
fALPHA_mat <- get_shannon(fDAT_temp, f_rare)
rm(fDAT_temp)

#Bacteria
bALPHA_summary <- cbind(rowMeans(bALPHA_mat), bDESIGN_maize)
levels(bALPHA_summary$Sample_type) <- c("root", "rhizosphere", "soil")
colnames(bALPHA_summary)[1] <- "ASV_diversity_all"

#Fungi
fALPHA_summary <- cbind(rowMeans(fALPHA_mat), fDESIGN_maize)
levels(fALPHA_summary$Sample_type) <- c("root", "rhizosphere", "soil")
colnames(fALPHA_summary)[1] <- "ASV_diversity_all"

#subseting data
bALPHA_maize <- bALPHA_summary[bALPHA_summary$Species == "maize",]
bALPHA_maize_root <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "root_maize",]
bALPHA_maize_rhizo <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "rhizosphere_maize",]
bALPHA_maize_soil <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "soil_maize",]

fALPHA_maize <- fALPHA_summary[fALPHA_summary$Species == "maize",]
fALPHA_maize_root <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "root_maize",]
fALPHA_maize_rhizo <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "rhizosphere_maize",]
fALPHA_maize_soil <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "soil_maize",]

```

## Anova statistics  


*Model: ~ Conditioning \* PCA_soil_chem_1*

```{r a-div bacteria ANOVA, echo=F, message=F, warning=F}

bALPHA_maize_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_maize)
bALPHA_maize_fit_aov <- summary(bALPHA_maize_fit)
rownames(bALPHA_maize_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_maize_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_maize_fit_aov, caption="Bacteria all compartments")

bALPHA_maize_root_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_maize_root)
bALPHA_maize_root_fit_aov <- summary(bALPHA_maize_root_fit)
rownames(bALPHA_maize_root_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_maize_root_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_maize_root_fit_aov, caption="Bacteria roots")

bALPHA_maize_rhizo_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_maize_rhizo)
bALPHA_maize_rhizo_fit_aov <- summary(bALPHA_maize_rhizo_fit)
rownames(bALPHA_maize_rhizo_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_maize_rhizo_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_maize_rhizo_fit_aov, caption="Bacteria rhizosphere")

bALPHA_maize_soil_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_maize_soil)
bALPHA_maize_soil_fit_aov <- summary(bALPHA_maize_soil_fit)
rownames(bALPHA_maize_soil_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_maize_soil_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_maize_soil_fit_aov, caption="Bacteria soil")

```

```{r a-div fungi ANOVA, echo=F, message=F, warning=F}

#Fungi
fALPHA_maize_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=fALPHA_maize)
fALPHA_maize_fit_aov <- summary(fALPHA_maize_fit)
rownames(fALPHA_maize_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_maize_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(fALPHA_maize_fit_aov, caption="Fungi all compartments")

fALPHA_maize_root_fit <- aov(ASV_diversity_all ~ 
                               Conditioning * PCA_soil_chem_1, data=fALPHA_maize_root)
fALPHA_maize_root_fit_aov <- summary(fALPHA_maize_root_fit)
rownames(fALPHA_maize_root_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_maize_root_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_maize_root_fit_aov, caption="Fungi roots")

fALPHA_maize_rhizo_fit <- aov(ASV_diversity_all ~ 
                                Conditioning * PCA_soil_chem_1, data=fALPHA_maize_rhizo)
fALPHA_maize_rhizo_fit_aov <- summary(fALPHA_maize_rhizo_fit)
rownames(fALPHA_maize_rhizo_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_maize_rhizo_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_maize_rhizo_fit_aov, caption="Fungi rhizosphere")

fALPHA_maize_soil_fit <- aov(ASV_diversity_all ~ 
                               Conditioning * PCA_soil_chem_1, data=fALPHA_maize_soil)
fALPHA_maize_soil_fit_aov <- summary(fALPHA_maize_soil_fit)
rownames(fALPHA_maize_soil_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_maize_soil_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_maize_soil_fit_aov, caption="Fungi soil")

```

\pagebreak

### Figure 5 | Shannon diversity by sampletype and conditioning  

```{r a-div maize plot, echo=F, message=F, warning=F, eval=T, fig.height=5, fig.width=10}

#create new color string for genotype (same color as for conditioning)
level_cols_Genotype <- level_cols_Conditioning
names(level_cols_Genotype) <- c("bx1", "W22")

#function plot a-div
plotALPHA <- function(data, title="", label=""){
  p <- ggplot(data, aes(y=ASV_diversity_all, x=Sample_type, fill=Genotype, col=PCA_soil_chem_1 )) +
        geom_jitter(aes(group=Sample_type_Conditioning), 
                    size=2, position=position_jitterdodge(jitter.width=0.2, dodge.width=0.8))+
        geom_boxplot(position=position_dodge2(width=0.75, preserve="single"), outlier.shape=NA, alpha=3/4) +
        theme_bw() +
        scale_fill_manual(labels = c(W22 = "WT", bx1 = "bx1"), values=level_cols_Genotype)+
        ylab("Shanon Diversity") +
        xlab("") +
        labs(title=title) +
        ylim(0, max(data$ASV_diversity_all*1.2)) +
        stat_summary(geom='text', label=label, fun.y=max, vjust=-1, position=position_dodge(width = 0.75))+
        theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
              legend.margin = margin(b =  0, unit='cm'),
              legend.box.margin = margin(b = -10, unit = "pt"))

  return(p)
}

#make plots
bplot <- plotALPHA(bALPHA_maize, "Bacteria: Alpha diversity")
fplot <- plotALPHA(fALPHA_maize, "Fungi: Alpha diversity")

#combine plots
legend <- get_legend(bplot + labs(color="Soil chemistry PC1", fill="Genotype"))
plot <- plot_grid(bplot + theme(legend.position="none"), 
                  fplot + theme(legend.position="none"), 
                  labels=c("A","B"), align = "hv", nrow=1)
plot <- plot_grid(plot, legend, nrow=2, rel_heights=c(1, .1))
print(plot)

```

*Conclusion:* The chemical field gradient influenced the alpha diversity of the microbial communities. Root and rhizosphere bacterial communities have higher Shannon diversity in BX+ compared to BX- conditions (weakly significant). Rhizosphere fungal communities have higher Shannon diversity in BX+ compared to BX- conditions (significant). 

\pagebreak

## Correlation analysis: Alpha diversity ~ soil chemical gradient

Is there a correlation between the alpha diversity and the soil chemical gradient? We tested for each sampletype.

### Figure 6.1 | Bacteria: Correlation alpha diversity ~ soil chemical gradient 

```{r a-div bacteria gradient correlation test and plot, echo=F, message=F, warning=F, eval=T, fig.height=6, fig.width=18} 

#correlation tests
cor2 <- cor.test(bALPHA_maize_root$ASV_diversity_all, bALPHA_maize_root$PCA_soil_chem_1)
cor3 <- cor.test(bALPHA_maize_rhizo$ASV_diversity_all, bALPHA_maize_rhizo$PCA_soil_chem_1)
cor6 <- cor.test(bALPHA_maize_soil$ASV_diversity_all, bALPHA_maize_soil$PCA_soil_chem_1)

#function correlation plot
plot_cor <- function(bdata, x, title="", cor){
  if(x == "PCA_soil_chem_1"){xlabel <- "PCA 1"}
  if(x == "PCA_soil_chem_4"){xlabel <- "PCA 4"}
  plot <- ggplot(bdata, aes(y=ASV_diversity_all, x=get(x)))+
      geom_jitter(width = 0, size=3) + 
      geom_smooth(se=F, method="lm", color="red")+
      theme_bw()+
      labs(title = title, subtitle = paste("cor:", round(cor$estimate,4), "p:", round(cor$p.value,4)))+
      xlab(xlabel)+
      ylab("Shannon Diversity")
  return(plot)
}

#make plots
plot2 <- plot_cor(bALPHA_maize_root, "PCA_soil_chem_1", "Bacteria: Roots maize", cor2)
plot3 <- plot_cor(bALPHA_maize_rhizo, "PCA_soil_chem_1", "Bacteria: Rhizo maize", cor3)
plot6 <- plot_cor(bALPHA_maize_soil, "PCA_soil_chem_1", "Bacteria: Soil maize", cor6)

#combine and plot
plot <- plot_grid(plot2, plot3, plot6, labels=c("A","B","C"), align = "hv", nrow=1, ncol=3)
plot

```

### Figure 6.2 | Fungi: Correlation alpha diversity ~ soil chemical gradient 

```{r a-div fungi gradient correlation test, echo=F, message=F, warning=F, eval=T, fig.height=6, fig.width=18} 

#correlation tests
cor2 <- cor.test(fALPHA_maize_root$ASV_diversity_all, fALPHA_maize_root$PCA_soil_chem_1)
cor3 <- cor.test(fALPHA_maize_rhizo$ASV_diversity_all, fALPHA_maize_rhizo$PCA_soil_chem_1)
cor6 <- cor.test(fALPHA_maize_soil$ASV_diversity_all, fALPHA_maize_soil$PCA_soil_chem_1)

#make plots
plot2 <- plot_cor(fALPHA_maize_root, "PCA_soil_chem_1", "Fungi: Roots maize", cor2)
plot3 <- plot_cor(fALPHA_maize_rhizo, "PCA_soil_chem_1", "Fungi: Rhizo maize", cor3)
plot6 <- plot_cor(fALPHA_maize_soil, "PCA_soil_chem_1", "Fungi: Soil maize", cor6)

#combine and plot
plot <- plot_grid(plot2, plot3, plot6, labels=c("A","B","C"), align = "hv", nrow=1, ncol=3)
plot

```

\pagebreak

# Analysis of BX conditioning effects - Beta diversity

We measured the beta diversity using Bray-Curtis distances. We asked for conditioning effects in each compartment by using PERMANOVA (function 'adonis2()' of the package vegan) and CAP ordination.

## PERMANOVA by sampletype  

*Model: ~ Conditioning \* PCA_soil_chem_1*  

```{r PERMANOVA bacteria conditioning, warning=F, echo=F, eval=T, message=F}

#Bacteria
#root
bDAT_rare_maize_root_dist_paov <- adonis2(bDAT_rare_maize_root_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_maize_root, permutations=999)
rownames(bDAT_rare_maize_root_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_maize_root_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  
pander(bDAT_rare_maize_root_dist_paov, caption="Bacteria: roots maize")

#rhizo
bDAT_rare_maize_rhizo_dist_paov <- adonis2(bDAT_rare_maize_rhizo_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_maize_rhizo, permutations=999)
rownames(bDAT_rare_maize_rhizo_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_maize_rhizo_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  
pander(bDAT_rare_maize_rhizo_dist_paov, caption="Bacteria: rhizo maize")

#soil
bDAT_rare_maize_soil_dist_paov <- adonis2(bDAT_rare_maize_soil_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_maize_soil, permutations=999)
rownames(bDAT_rare_maize_soil_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_maize_soil_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  
pander(bDAT_rare_maize_soil_dist_paov, caption="Bacteria: soil maize")

```

```{r PERMANOVA Fungi conditioning, warning=F, echo=F, eval=T, message=F}

#Fungi
#root
fDAT_rare_maize_root_dist_paov <- adonis2(fDAT_rare_maize_root_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                          data=fDESIGN_maize_root, permutations=999)
rownames(fDAT_rare_maize_root_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_maize_root_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  
pander(fDAT_rare_maize_root_dist_paov, caption="Fungi: roots maize")

#rhizo
fDAT_rare_maize_rhizo_dist_paov <- adonis2(fDAT_rare_maize_rhizo_dist ~ 
                                             Conditioning * PCA_soil_chem_1, 
                                           data=fDESIGN_maize_rhizo, permutations=999)
rownames(fDAT_rare_maize_rhizo_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_maize_rhizo_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  
pander(fDAT_rare_maize_rhizo_dist_paov, caption="Fungi: rhizo maize")

#soil
fDAT_rare_maize_soil_dist_paov <- adonis2(fDAT_rare_maize_soil_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                          data=fDESIGN_maize_soil, permutations=999)
pander(fDAT_rare_maize_soil_dist_paov, caption="Fungi: soil maize")
rownames(fDAT_rare_maize_soil_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_maize_soil_dist_paov)[3] <- "Conditioning : Soil chemistry PC1"  

```

\pagebreak 

## CAP by sampletype  

*Model: ~ Conditioning \* PCA_soil_chem_1* 

```{r subsetting phyloseq objects, echo=F, warning=F, message=F, eval=T}

#Bacteria
#bPHYSEQ_rare_maize
bPHYSEQ_rare_maize_root <- prune_samples(sample_data(bPHYSEQ_rare_maize)$Sample_type=="root", bPHYSEQ_rare_maize)
bPHYSEQ_rare_maize_rhizo <- prune_samples(sample_data(bPHYSEQ_rare_maize)$Sample_type=="rhizosphere", bPHYSEQ_rare_maize)
bPHYSEQ_rare_maize_soil <- prune_samples(sample_data(bPHYSEQ_rare_maize)$Sample_type=="soil", bPHYSEQ_rare_maize)

#Fungi
#fPHYSEQ_rare_maize
fPHYSEQ_rare_maize_root <- prune_samples(sample_data(fPHYSEQ_rare_maize)$Sample_type=="root", fPHYSEQ_rare_maize)
fPHYSEQ_rare_maize_rhizo <- prune_samples(sample_data(fPHYSEQ_rare_maize)$Sample_type=="rhizosphere", fPHYSEQ_rare_maize)
fPHYSEQ_rare_maize_soil <- prune_samples(sample_data(fPHYSEQ_rare_maize)$Sample_type=="soil", fPHYSEQ_rare_maize)

```

```{r CAPs, echo=F, warning=F, message=F, eval=T}

#create new color string for genotype (same color as for conditioning)
level_cols_Genotype <- level_cols_Conditioning
names(level_cols_Genotype) <- c("bx1", "W22")

#function plot CAP
CAP_plot <- function(bPHY_object, CAP_object, title="", stats=""){
  p0 <- phyloseq::plot_ordination(bPHY_object, CAP_object, color='Genotype', axes = c(1,2), title = title)
  p0$layers <- p0$layers[-1]
  p0 <- p0 + geom_point(aes(size = PCA_soil_chem_1 ), stroke=1, alpha=3/4) 
  p0 <- p0 + stat_ellipse(geom = "polygon", type="t", alpha=0.2, aes(fill=Genotype))
  p0 <- p0 + scale_fill_manual(labels = c(W22 = "WT", bx1 = "bx1"), values=level_cols_Genotype)
  p0 <- p0 + scale_color_manual(labels = c(W22 = "WT", bx1 = "bx1"), values=level_cols_Genotype)
  #p0 <- p0 + geom_text(label = bDESIGN_maize_root$PCA_soil_chem_1, size = 3, vjust=2)
  p0 <- p0 + theme_bw(base_size = 12)
  p0 <- p0 + theme(legend.position="none")
  p0 <- p0 + scale_x_reverse()
  p0 <- p0 + labs(subtitle = stats)
  return(p0)
}

#maize root bacteria
bPHYSEQ_CAP_bray_maize_root <- phyloseq::ordinate(bPHYSEQ_rare_maize_root, method="CAP", distance="bray",  
                                                   formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_maize_root) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_maize_root, permutations=999) #Anova
bPHYSEQ_CAP_bray_maize_root_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmroot=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_root_maize <- CAP_plot(bPHYSEQ_rare_maize_root, 
                            bPHYSEQ_CAP_bray_maize_root, "Maize root bacteria", 
                            bPHYSEQ_CAP_bray_maize_root_stats)


#maize rhizo bacteria
bPHYSEQ_CAP_bray_maize_rhizo <- phyloseq::ordinate(bPHYSEQ_rare_maize_rhizo, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_maize_rhizo) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_maize_rhizo, permutations=999) #Anova
bPHYSEQ_CAP_bray_maize_rhizo_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmrhizo=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_rhizo_maize <- CAP_plot(bPHYSEQ_rare_maize_rhizo, 
                            bPHYSEQ_CAP_bray_maize_rhizo, "Maize rhizo bacteria", 
                            bPHYSEQ_CAP_bray_maize_rhizo_stats)


#maize soil bacteria
bPHYSEQ_CAP_bray_maize_soil <- phyloseq::ordinate(bPHYSEQ_rare_maize_soil, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_maize_soil) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_maize_soil, permutations=999) #Anova
bPHYSEQ_CAP_bray_maize_soil_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmsoil=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_soil_maize <- CAP_plot(bPHYSEQ_rare_maize_soil, 
                            bPHYSEQ_CAP_bray_maize_soil, "Maize soil bacteria", 
                            bPHYSEQ_CAP_bray_maize_soil_stats)




#maize root fungi
fPHYSEQ_CAP_bray_maize_root <- phyloseq::ordinate(fPHYSEQ_rare_maize_root, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
# stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_maize_root) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_maize_root, permutations=999) #Anova
fPHYSEQ_CAP_bray_maize_root_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmroot=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_root_maize <- CAP_plot(fPHYSEQ_rare_maize_root, 
                            fPHYSEQ_CAP_bray_maize_root, "Maize root fungi", 
                            fPHYSEQ_CAP_bray_maize_root_stats)


#maize rhizo fungi
fPHYSEQ_CAP_bray_maize_rhizo <- phyloseq::ordinate(fPHYSEQ_rare_maize_rhizo, method="CAP", distance="bray",  
                                                   formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_maize_rhizo) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_maize_rhizo, permutations=999) #Anova
fPHYSEQ_CAP_bray_maize_rhizo_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmrhizo=1),
                                            "% of variance, P = ",
                                            format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_rhizo_maize <- CAP_plot(fPHYSEQ_rare_maize_rhizo, 
                             fPHYSEQ_CAP_bray_maize_rhizo, "Maize rhizo fungi", 
                             fPHYSEQ_CAP_bray_maize_rhizo_stats)


#maize soil fungi
fPHYSEQ_CAP_bray_maize_soil <- phyloseq::ordinate(fPHYSEQ_rare_maize_soil, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_maize_soil) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_maize_soil, permutations=999) #Anova
fPHYSEQ_CAP_bray_maize_soil_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmsoil=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_soil_maize <- CAP_plot(fPHYSEQ_rare_maize_soil, 
                            fPHYSEQ_CAP_bray_maize_soil, "Maize soil fungi", 
                            fPHYSEQ_CAP_bray_maize_soil_stats)

#legend
for_legend <- phyloseq::plot_ordination(fPHYSEQ_rare_maize_soil, fPHYSEQ_CAP_bray_maize_soil, color="Genotype", axes = c(1,2))
for_legend <- for_legend + geom_point(aes(size = PCA_soil_chem_1 ), stroke=1, alpha=3/4)
for_legend <- for_legend + scale_color_manual(labels = c(W22 = "WT", bx1 = "bx1"), values=level_cols_Genotype)
for_legend <- for_legend + guides(color = guide_legend(order = 1),size = guide_legend(order = 2))
   
```

### Figure 7 | CAP of maize (by sampletype) 

```{r multiplot CAP, echo=F, warning=F, message=F, eval=T, fig.height=14, fig.width=18}

#combine and plot
CAP_maize_sample_type <- plot_grid(bCAP_root_maize, bCAP_rhizo_maize, bCAP_soil_maize,
                                   fCAP_root_maize, fCAP_rhizo_maize, fCAP_soil_maize,
                                   align='vh', rel_widths=c(1,1,1,1,1,1),
                                   labels=c("A","B","C","D","F","G"), hjust=-1, ncol=3 )

legend <- get_legend(for_legend + theme(legend.position="bottom") + labs(size="Soil chemistry PC1"))
CAP_multi_maize_sample_type <- plot_grid(CAP_maize_sample_type, legend, nrow=2, rel_heights = c(1, 0.2))

print(CAP_multi_maize_sample_type)

```

*Conclusion:* The chemical gradient affected the bacterial communities more strongly (sig. in all compartments) compared to BX conditioning. A significant BX effect was found in maize roots for bacteria. Rhizospere and soil are not affected. The chemical gradient affected the fungal communities less than the bacteria (lower R2 and only in root and rhizosphere). The BX conditioning however was more pronounced, being significant in roots and rhizosphere, but not soil samples.

\pagebreak
