---
title: "Wheat growth"
author: "Jan WÃ¤lchli, Selma Cadot and Klaus Schlaeppi"
output:
  pdf_document: 
    toc: yes
    toc_depth: 3
---
   
```{r setup, include=FALSE, , echo=F}

##clear the object from memory
rm(list=ls())

#knitr settings
knitr::opts_chunk$set(echo = TRUE, fig.align = "center")
options(tinytex.verbose = TRUE)

#set seed
set.seed(100)

```

```{r libraries, echo=F, message=F, warning=F}

#set source to file location
if (!require(rstudioapi)){install.packages("rstudioapi")}; library(rstudioapi)
setwd(dirname(getActiveDocumentContext()$path))

#paths
paths <- list(interim = c("../1_data/interim"))

#load functions
source("functions/libraries.R")
source("functions/variability_table.R")

#installs (if necessary) and loads libraries
libraries()

#ggplot settings
theme_set(theme_bw(base_size = 12))

```

```{r paths, echo=F, message=F, warning=F}

#set wd to RDS files
setwd(paths$interim)

#import RDS files
bDESIGN <- readRDS("bDESIGN.RDS")
fDESIGN <- readRDS("fDESIGN.RDS")

bDAT <- readRDS("bDAT.RDS")
fDAT <- readRDS("fDAT.RDS")
b_rare <- readRDS("b_rare.RDS")
f_rare <- readRDS("f_rare.RDS")
bDAT_rare <- readRDS("bDAT_rare.RDS")
fDAT_rare <- readRDS("fDAT_rare.RDS")

bTAX <- readRDS("bTAX.RDS")
fTAX <- readRDS("fTAX.RDS")

level_cols_Sample_type <- readRDS("level_cols_Sample_type.RDS")
level_cols_Conditioning <- readRDS("level_cols_Conditioning.RDS")

```

```{r subsetting, echo=F, warning=F, message=F, eval=T, }


#Bacteria
#subset DESIGN
bDESIGN_wheat <- droplevels(bDESIGN[bDESIGN$Species=="wheat", ])
rownames(bDESIGN_wheat) <- bDESIGN_wheat$bprimer
bDESIGN_wheat_root <- droplevels(bDESIGN_wheat[bDESIGN_wheat$Sample_type == "root",])
bDESIGN_wheat_rhizo <- droplevels(bDESIGN_wheat[bDESIGN_wheat$Sample_type == "rhizosphere",])
bDESIGN_wheat_soil <- droplevels(bDESIGN_wheat[bDESIGN_wheat$Sample_type == "soil",])

#subset DATA_rare
bDAT_rare_wheat <- bDAT_rare[ ,rownames(bDESIGN_wheat)]
bDAT_rare_wheat <- bDAT_rare_wheat[rowSums(bDAT_rare_wheat) > 0,]  # removal of rows with 0 values
bDAT_rare_wheat_root <- bDAT_rare_wheat[ ,rownames(bDESIGN_wheat_root)]
bDAT_rare_wheat_root <- bDAT_rare_wheat_root[rowSums(bDAT_rare_wheat_root) > 0,]  # removal of rows with 0 values
bDAT_rare_wheat_rhizo <- bDAT_rare_wheat[ ,rownames(bDESIGN_wheat_rhizo)]
bDAT_rare_wheat_rhizo <- bDAT_rare_wheat_rhizo[rowSums(bDAT_rare_wheat_rhizo) > 0,]  # removal of rows with 0 values
bDAT_rare_wheat_soil <- bDAT_rare_wheat[ ,rownames(bDESIGN_wheat_soil)]
bDAT_rare_wheat_soil <- bDAT_rare_wheat_soil[rowSums(bDAT_rare_wheat_soil) > 0,]  # removal of rows with 0 values

#make PHYLOSEQ
bPHYSEQ_rare_wheat <- phyloseq(sample_data(bDESIGN_wheat),
                               otu_table(bDAT_rare_wheat, taxa_are_rows=T),
                               tax_table(as.matrix(bTAX[rownames(bDAT_rare_wheat),])) )

#Fungi
#subset DESIGN
fDESIGN_wheat <- droplevels(fDESIGN[fDESIGN$Species=="wheat", ])
rownames(fDESIGN_wheat) <- fDESIGN_wheat$fprimer
fDESIGN_wheat_root <- droplevels(fDESIGN_wheat[fDESIGN_wheat$Sample_type == "root",])
fDESIGN_wheat_rhizo <- droplevels(fDESIGN_wheat[fDESIGN_wheat$Sample_type == "rhizosphere",])
fDESIGN_wheat_soil <- droplevels(fDESIGN_wheat[fDESIGN_wheat$Sample_type == "soil",])

#subset DATA_rare
fDAT_rare_wheat <- fDAT_rare[ ,rownames(fDESIGN_wheat)]
fDAT_rare_wheat <- fDAT_rare_wheat[rowSums(fDAT_rare_wheat) > 0,]  # removal of rows with 0 values
fDAT_rare_wheat_root <- fDAT_rare_wheat[ ,rownames(fDESIGN_wheat_root)]
fDAT_rare_wheat_root <- fDAT_rare_wheat_root[rowSums(fDAT_rare_wheat_root) > 0,]  # removal of rows with 0 values
fDAT_rare_wheat_rhizo <- fDAT_rare_wheat[ ,rownames(fDESIGN_wheat_rhizo)]
fDAT_rare_wheat_rhizo <- fDAT_rare_wheat_rhizo[rowSums(fDAT_rare_wheat_rhizo) > 0,]  # removal of rows with 0 values
fDAT_rare_wheat_soil <- fDAT_rare_wheat[ ,rownames(fDESIGN_wheat_soil)]
fDAT_rare_wheat_soil <- fDAT_rare_wheat_soil[rowSums(fDAT_rare_wheat_soil) > 0,]  # removal of rows with 0 values

#make PHYLOSEQ
fPHYSEQ_rare_wheat <- phyloseq(sample_data(fDESIGN_wheat),
                               otu_table(fDAT_rare_wheat, taxa_are_rows=T),
                               tax_table(as.matrix(fTAX[rownames(fDAT_rare_wheat),])) )

```

\pagebreak 

# Data analysis - Wheat growth

Here we analysed the microbiome data which has been collected after wheat has grown.

# Soil chemistry PC1

We know from previous analyses that we have a soil-chemical gradient (soil chemistry PC1) along the field. We tested if the microbiome changes along soil chemistry PC1. We tested this effect with a PERMANOVA and illustrated it with a PCoA.

## PERMANOVA

*Model: ~ Sample_type + Soil Chemistry PC1*

```{r bray curtis distances, warning=F, echo=F, eval=T}

#Bacteria
bDAT_rare_wheat_dist <- vegdist(t(bDAT_rare_wheat), method="bray")
bDAT_rare_wheat_root_dist <- vegdist(t(bDAT_rare_wheat_root), method="bray")
bDAT_rare_wheat_rhizo_dist <- vegdist(t(bDAT_rare_wheat_rhizo), method="bray")
bDAT_rare_wheat_soil_dist <- vegdist(t(bDAT_rare_wheat_soil), method="bray")

bDAT_rare_wheat_dist_paov <- adonis2(bDAT_rare_wheat_dist ~ Sample_type + PCA_soil_chem_1, data=bDESIGN_wheat, permutations=999)
rownames(bDAT_rare_wheat_dist_paov)[1] <- "Sampletype"
rownames(bDAT_rare_wheat_dist_paov)[2] <- "Soil chemistry PC1"  
pander(bDAT_rare_wheat_dist_paov[1:3,c(3,5)], caption="Bacteria")

#Fungi
fDAT_rare_wheat_dist <- vegdist(t(fDAT_rare_wheat), method="bray")
fDAT_rare_wheat_root_dist <- vegdist(t(fDAT_rare_wheat_root), method="bray")
fDAT_rare_wheat_rhizo_dist <- vegdist(t(fDAT_rare_wheat_rhizo), method="bray")
fDAT_rare_wheat_soil_dist <- vegdist(t(fDAT_rare_wheat_soil), method="bray")

#fDAT_rare_wheat_dist_paov <- adonis2(fDAT_rare_wheat_dist ~ Sample_type + pos_long + pos_lat, data=fDESIGN_wheat, permutations=999)
fDAT_rare_wheat_dist_paov <- adonis2(fDAT_rare_wheat_dist ~ Sample_type + PCA_soil_chem_1, data=fDESIGN_wheat, permutations=999)
rownames(fDAT_rare_wheat_dist_paov)[1] <- "Sampletype"
rownames(fDAT_rare_wheat_dist_paov)[2] <- "Soil chemistry PC1"  
pander(fDAT_rare_wheat_dist_paov[1:3,c(3,5)], caption="Fungi")

```

\pagebreak 

## PCoA

We performed an unconstrained ordination with Bray-Curtis distances.

### Figure 12 | PCoA with Bray-Curtis 

```{r bacteria PCoA all samples, echo=F, warning=F, message=F, eval=T, fig.height=6, fig.width=11}

#Bacteria
#PCoA plot
bPHYSEQ_rare_wheat_PCoA_bray <- phyloseq::ordinate(bPHYSEQ_rare_wheat, method="PCoA", distance="bray")
p0 <- phyloseq::plot_ordination(bPHYSEQ_rare_wheat, bPHYSEQ_rare_wheat_PCoA_bray, shape="Sample_type", color="PCA_soil_chem_1", title = "Bacteria at wheat growth (PCoA, Bray-Curtis)", axes = c(1,2))
p0 <- p0 + geom_point(size=3)
p0 <- p0 + scale_shape_manual(values=c(16,17,15))
p0 <- p0 +  theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
                  legend.margin = margin(b =  0, unit='cm'),
                  legend.box.margin = margin(b = -10, unit = "pt"))
Fig_3A <- p0

#Fungi
#PCoA plot
fPHYSEQ_rare_wheat_PCoA_bray <- phyloseq::ordinate(fPHYSEQ_rare_wheat, method="PCoA", distance="bray")
p0 <- phyloseq::plot_ordination(fPHYSEQ_rare_wheat, fPHYSEQ_rare_wheat_PCoA_bray, shape="Sample_type", color="PCA_soil_chem_1", title = "Fungi at wheat growth (PCoA, Bray-Curtis)", axes = c(1,2))
p0 <- p0 + geom_point(size=3)
p0 <- p0 + scale_shape_manual(values=c(16,17,15)) 
p0 <- p0 +  theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
                  legend.margin = margin(b =  0, unit='cm'),
                  legend.box.margin = margin(b = -10, unit = "pt"))
Fig_3B <- p0

#combine plots (package:cowplot)
legend <- get_legend(Fig_3A + theme(legend.position="bottom") + labs(shape="Sampletype", col="Soil Chemistry PC1"))
p <- plot_grid(Fig_3A + theme(legend.position="none"),
               Fig_3B + theme(legend.position="none"),
                  labels=c(" "," "), align = "hv", nrow=1)

p <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(p)

```

\pagebreak 

## Correlation analysis: PCoA (microbiota) ~ Soil chemistry PC1

Do soil chemistry PC1 correlate with the beta diversity? We tested for the first two axes.

### Figure 13.1 | Bacteria: Correlation (microbiota) ~ Soil chemistry PC1

```{r correlation bacteria, warning=F, echo=F, eval=T, message=F, fig.height=6, fig.width=11}

#function to convert r to R2
r_to_R2 <- function(r){paste0((r^2)*100,"%")}

#function plot beta div - soil chem
plot_cor <- function(df, cor, PCoA_axis, ylabel, pos_field, xlabel){
  plot <- ggplot(df, aes(y=get(PCoA_axis), x=get(pos_field))) +
            geom_jitter(width = 0.1, size=3, aes(shape=Sample_type)) + 
            stat_smooth(method = "lm", se = FALSE, fullrange = TRUE, colour = "dimgrey", alpha = 3/4) +
            theme_bw(base_size = 12) +
            theme(legend.position="none") +
            xlab(xlabel) +
            ylab(ylabel) +
            labs(title = paste("R2:", r_to_R2(round(cor$estimate,2)), "p:", formatC(cor$p.value, format = "g", digits = 2)))
  return(plot)
}

#get PCoA dataframe
bPHYSEQ_PCoA_bray_df <- phyloseq::plot_ordination(bPHYSEQ_rare_wheat, bPHYSEQ_rare_wheat_PCoA_bray, justDF = T)

#correlation tests (PCoA ~ position)
bPHYSEQ_PCoA_bray_df_A1_chem1 <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.1, bPHYSEQ_PCoA_bray_df$PCA_soil_chem_1)
bPHYSEQ_PCoA_bray_df_A2_chem1 <- cor.test(bPHYSEQ_PCoA_bray_df$Axis.2, bPHYSEQ_PCoA_bray_df$PCA_soil_chem_1)

#make plots
p1 <- plot_cor(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A1_chem1, "Axis.1", "PCoA Axis 1", "PCA_soil_chem_1", "Soil chemistry PC1")
p2 <- plot_cor(bPHYSEQ_PCoA_bray_df, bPHYSEQ_PCoA_bray_df_A2_chem1, "Axis.2", "PCoA Axis 2", "PCA_soil_chem_1", "Soil chemistry PC1")

#combine plots
legend <- get_legend(p1 + theme(legend.position="bottom") + labs(shape="Sampletype"))
p <- plot_grid(p1 + theme(legend.position="none"),
               p2 + theme(legend.position="none"),
               labels=c(" "," "), align = "hv", nrow=1)

bCORplot <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(bCORplot)

```



### Figure 13.2 | Fungi: Correlation (microbiota) ~ Soil chemistry PC1

```{r correlation fungi, warning=F, echo=F, eval=T, message=F, fig.height=6, fig.width=11}

#get PCoA dataframe
fPHYSEQ_PCoA_bray_df <- phyloseq::plot_ordination(fPHYSEQ_rare_wheat, fPHYSEQ_rare_wheat_PCoA_bray, justDF = T)

#correlation tests (PCoA ~ position)
fPHYSEQ_PCoA_bray_df_A1_chem1 <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.1, fPHYSEQ_PCoA_bray_df$PCA_soil_chem_1)
fPHYSEQ_PCoA_bray_df_A2_chem1 <- cor.test(fPHYSEQ_PCoA_bray_df$Axis.2, fPHYSEQ_PCoA_bray_df$PCA_soil_chem_1)

#make plots
p1 <- plot_cor(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A1_chem1, "Axis.1", "PCoA Axis 1", "PCA_soil_chem_1", "Soil chemistry PC1")
p2 <- plot_cor(fPHYSEQ_PCoA_bray_df, fPHYSEQ_PCoA_bray_df_A2_chem1, "Axis.2", "PCoA Axis 2", "PCA_soil_chem_1", "Soil chemistry PC1")

#combine plots
legend <- get_legend(p1 + theme(legend.position="bottom") + labs(shape="Sampletype"))
p <- plot_grid(p1 + theme(legend.position="none"),
               p2 + theme(legend.position="none"),
               labels=c(" "," "), align = "hv", nrow=1)

fCORplot <- plot_grid(p, legend, nrow=2, rel_heights=c(1, .1))
print(fCORplot)

```

**Conclusion:** The microbiome is influenced by soil chemistry PC1. To take soil chemistry PC1 into account, we included it as independent variable in all following models.

\pagebreak

# Analysis of BX conditioning effects - Alpha diversity

How did BX conditioning affect the microbiome? We compared the root, rhizosphere and soil microbiomes of wheat whether the microbial communities would differ between the conditioning treatments (WT, *bx1*). To measure the alpha diversity, we rarefied the data to the sequencing depth of `r b_rare` (bacteria) and `r f_rare` (fungi). Then we calculated Shannon diversity in each sample. This was repeated 100 times and the mean value from the 100 iterations was taken for statistical analysis between the different samples.  

```{r a-div shannon, warning=F, echo=F, message=F, eval=T}

#function to calculate Shannon
get_shannon <- function(DAT, rare){
  #get shannon diversity
  diversity_all_DAT_mat <- c()
  for (i in 1:100){
    #generate rarefied community
    temp_mat <- t(rrarefy(t(DAT), rare))
    shannon <- vegan::diversity(temp_mat, index="shannon", MARGIN=2)
    d <- exp(shannon)
    diversity_all_DAT_mat <- cbind(diversity_all_DAT_mat, d)
    }
    return(diversity_all_DAT_mat)
}

#shannon div
bDAT_temp <- bDAT[, rownames(bDESIGN_wheat)]
bDAT_temp <- bDAT_temp[rowSums(bDAT_temp) > 0,]  #removal of rows with 0 values
bALPHA_mat <- get_shannon(bDAT_temp, b_rare)
rm(bDAT_temp)

fDAT_temp <- fDAT[, rownames(fDESIGN_wheat)]
fDAT_temp <- fDAT_temp[rowSums(fDAT_temp) > 0,]  # emoval of rows with 0 values
fALPHA_mat <- get_shannon(fDAT_temp, f_rare)
rm(fDAT_temp)

#Bacteria
bALPHA_summary <- cbind(rowMeans(bALPHA_mat), bDESIGN_wheat)
levels(bALPHA_summary$Sample_type) <- c("root", "rhizosphere", "soil")
colnames(bALPHA_summary)[1] <- "ASV_diversity_all"

#Fungi
fALPHA_summary <- cbind(rowMeans(fALPHA_mat), fDESIGN_wheat)
levels(fALPHA_summary$Sample_type) <- c("root", "rhizosphere", "soil")
colnames(fALPHA_summary)[1] <- "ASV_diversity_all"

#subseting data
bALPHA_wheat <- bALPHA_summary[bALPHA_summary$Species == "wheat",]
bALPHA_wheat_root <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "root_wheat",]
bALPHA_wheat_rhizo <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "rhizosphere_wheat",]
bALPHA_wheat_soil <- bALPHA_summary[bALPHA_summary$Sample_type_Species == "soil_wheat",]

fALPHA_wheat <- fALPHA_summary[fALPHA_summary$Species == "wheat",]
fALPHA_wheat_root <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "root_wheat",]
fALPHA_wheat_rhizo <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "rhizosphere_wheat",]
fALPHA_wheat_soil <- fALPHA_summary[fALPHA_summary$Sample_type_Species == "soil_wheat",]

```

## Anova statistics  


*Model: ~ Conditioning \* Soil chemistry PC1*

```{r a-div bacteria ANOVA, echo=F, message=F, warning=F}

bALPHA_wheat_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_wheat)
bALPHA_wheat_fit_aov <- summary(bALPHA_wheat_fit)
rownames(bALPHA_wheat_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_wheat_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_wheat_fit_aov, caption="Bacteria all compartments")

bALPHA_wheat_root_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_wheat_root)
bALPHA_wheat_root_fit_aov <- summary(bALPHA_wheat_root_fit)
rownames(bALPHA_wheat_root_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_wheat_root_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_wheat_root_fit_aov, caption="Bacteria roots")

bALPHA_wheat_rhizo_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_wheat_rhizo)
bALPHA_wheat_rhizo_fit_aov <- summary(bALPHA_wheat_rhizo_fit)
rownames(bALPHA_wheat_rhizo_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_wheat_rhizo_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_wheat_rhizo_fit_aov, caption="Bacteria rhizosphere")

bALPHA_wheat_soil_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=bALPHA_wheat_soil)
bALPHA_wheat_soil_fit_aov <- summary(bALPHA_wheat_soil_fit)
rownames(bALPHA_wheat_soil_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(bALPHA_wheat_soil_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(bALPHA_wheat_soil_fit_aov, caption="Bacteria soil")

```

```{r a-div fungi ANOVA, echo=F, message=F, warning=F}

#Fungi
fALPHA_wheat_fit <- aov(ASV_diversity_all ~ 
                          Conditioning * PCA_soil_chem_1, data=fALPHA_wheat)
fALPHA_wheat_fit_aov <- summary(fALPHA_wheat_fit)
rownames(fALPHA_wheat_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_wheat_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1"  
pander(fALPHA_wheat_fit_aov, caption="Fungi all compartments")

fALPHA_wheat_root_fit <- aov(ASV_diversity_all ~ 
                               Conditioning * PCA_soil_chem_1, data=fALPHA_wheat_root)
fALPHA_wheat_root_fit_aov <- summary(fALPHA_wheat_root_fit)
rownames(fALPHA_wheat_root_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_wheat_root_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_wheat_root_fit_aov, caption="Fungi roots")

fALPHA_wheat_rhizo_fit <- aov(ASV_diversity_all ~ 
                                Conditioning * PCA_soil_chem_1, data=fALPHA_wheat_rhizo)
fALPHA_wheat_rhizo_fit_aov <- summary(fALPHA_wheat_rhizo_fit)
rownames(fALPHA_wheat_rhizo_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_wheat_rhizo_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_wheat_rhizo_fit_aov, caption="Fungi rhizosphere")

fALPHA_wheat_soil_fit <- aov(ASV_diversity_all ~ 
                               Conditioning * PCA_soil_chem_1, data=fALPHA_wheat_soil)
fALPHA_wheat_soil_fit_aov <- summary(fALPHA_wheat_soil_fit)
rownames(fALPHA_wheat_soil_fit_aov[[1]])[2] <- "Soil chemistry PC1"
rownames(fALPHA_wheat_soil_fit_aov[[1]])[3] <- "Conditioning : Soil chemistry PC1" 
pander(fALPHA_wheat_soil_fit_aov, caption="Fungi soil")

```

\pagebreak

### Figure 14 | Shannon diversity by sampletype and conditioning  

```{r a-div wheat plot, echo=F, message=F, warning=F, eval=T, fig.height=5, fig.width=10}

#function plot a-div
plotALPHA <- function(data, title="", label=""){
  p <- ggplot(data, aes(y=ASV_diversity_all, x=Sample_type, fill=Conditioning, col=PCA_soil_chem_1 )) +
        geom_boxplot(position=position_dodge2(width=0.75, preserve="single"), outlier.shape=NA, alpha=3/4) +
        geom_jitter(aes(group=Sample_type_Conditioning), 
                    size=2, position=position_jitterdodge(jitter.width=0.2, dodge.width=0.8))+
        theme_bw() +
        scale_fill_manual(values=level_cols_Conditioning)+
        ylab("Shanon Diversity") +
        xlab("") +
        labs(title=title) +
        ylim(0, max(data$ASV_diversity_all*1.2)) +
        stat_summary(geom='text', label=label, fun.y=max, vjust=-1, position=position_dodge(width = 0.75))+
        theme(legend.position = "top", legend.key.height = unit(.3, "cm"),
              legend.margin = margin(b =  0, unit='cm'),
              legend.box.margin = margin(b = -10, unit = "pt"))

  return(p)
}

#make plots
bplot <- plotALPHA(bALPHA_wheat, "Bacteria: Alpha diversity")
fplot <- plotALPHA(fALPHA_wheat, "Fungi: Alpha diversity")

#combine plots
legend <- get_legend(bplot + labs(color="Soil chemistry PC1", fill="Conditioning"))
plot <- plot_grid(bplot + theme(legend.position="none"), 
                  fplot + theme(legend.position="none"), 
                  align = "hv", nrow=1)
plot <- plot_grid(plot, legend, nrow=2, rel_heights=c(1, .1))
print(plot)

```

**Conclusion:** Soil chemistry PC1 influenced the alpha diversity of root fungal communities. They have higher Shannon diversity in WT compared to *bx1* conditions (significant). 

\pagebreak

## Correlation analysis: alpha diversity ~ soil chemistry PC1

Is there a correlation between the alpha diversity and soil chemistry PC1? We tested for each sampletype.

### Figure 15.1 | Bacteria: Correlation alpha diversity ~ soil chemistry PC1

```{r a-div bacteria soil chemistry PC1 correlation test and plot, echo=F, message=F, warning=F, eval=T, fig.height=6, fig.width=18} 

#correlation tests
cor2 <- cor.test(bALPHA_wheat_root$ASV_diversity_all, bALPHA_wheat_root$PCA_soil_chem_1)
cor3 <- cor.test(bALPHA_wheat_rhizo$ASV_diversity_all, bALPHA_wheat_rhizo$PCA_soil_chem_1)
cor6 <- cor.test(bALPHA_wheat_soil$ASV_diversity_all, bALPHA_wheat_soil$PCA_soil_chem_1)

#function correlation plot
plot_cor <- function(bdata, x, title="", cor){
  if(x == "PCA_soil_chem_1"){xlabel <- "Soil chemistry PC1"}
  #if(x == "PCA_soil_chem_4"){xlabel <- "Soil chemistry PC4"}
  plot <- ggplot(bdata, aes(y=ASV_diversity_all, x=get(x)))+
      geom_jitter(width = 0, size=3) + 
      geom_smooth(se=F, method="lm", color="black")+
      theme_bw()+
      labs(title = title, subtitle = paste("cor:", round(cor$estimate,4), "p:", round(cor$p.value,4)))+
      xlab(xlabel)+
      ylab("Shannon Diversity")
  return(plot)
}

#make plots
plot2 <- plot_cor(bALPHA_wheat_root, "PCA_soil_chem_1", "Bacteria: Roots wheat", cor2)
plot3 <- plot_cor(bALPHA_wheat_rhizo, "PCA_soil_chem_1", "Bacteria: Rhizo wheat", cor3)
plot6 <- plot_cor(bALPHA_wheat_soil, "PCA_soil_chem_1", "Bacteria: Soil wheat", cor6)

#combine and plot
plot <- plot_grid(plot2, plot3, plot6, labels=c("A","B","C"), align = "hv", nrow=1, ncol=3)
plot

```

### Figure 15.2 | Fungi: Correlation alpha diversity ~ soil chemistry PC1

```{r a-div fungi soil chemistry PC1 correlation test, echo=F, message=F, warning=F, eval=T, fig.height=6, fig.width=18} 

#correlation tests
cor2 <- cor.test(fALPHA_wheat_root$ASV_diversity_all, fALPHA_wheat_root$PCA_soil_chem_1)
cor3 <- cor.test(fALPHA_wheat_rhizo$ASV_diversity_all, fALPHA_wheat_rhizo$PCA_soil_chem_1)
cor6 <- cor.test(fALPHA_wheat_soil$ASV_diversity_all, fALPHA_wheat_soil$PCA_soil_chem_1)

#make plots
plot2 <- plot_cor(fALPHA_wheat_root, "PCA_soil_chem_1", "Fungi: Roots wheat", cor2)
plot3 <- plot_cor(fALPHA_wheat_rhizo, "PCA_soil_chem_1", "Fungi: Rhizo wheat", cor3)
plot6 <- plot_cor(fALPHA_wheat_soil, "PCA_soil_chem_1", "Fungi: Soil wheat", cor6)

#combine and plot
plot <- plot_grid(plot2, plot3, plot6, labels=c("A","B","C"), align = "hv", nrow=1, ncol=3)
plot

```

\pagebreak

# Analysis of BX conditioning effects - Beta diversity

We measured the beta diversity using Bray-Curtis distances. We asked for conditioning effects in each compartment by using PERMANOVA (function 'adonis2()' of the package vegan) and CAP ordination.

## PERMANOVA by sampletype  

*Model: ~ Conditioning \* Soil chemistry PC1*  

#### Bacteria

```{r PERMANOVA bacteria conditioning, warning=F, echo=F, eval=T, message=F}

### root
bDAT_rare_wheat_root_dist_paov <- adonis2(bDAT_rare_wheat_root_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_wheat_root, permutations=999)
rownames(bDAT_rare_wheat_root_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_wheat_root_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(bDAT_rare_wheat_root_dist_paov, caption="Bacteria: roots wheat")

### rhizo
bDAT_rare_wheat_rhizo_dist_paov <- adonis2(bDAT_rare_wheat_rhizo_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_wheat_rhizo, permutations=999)
rownames(bDAT_rare_wheat_rhizo_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_wheat_rhizo_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(bDAT_rare_wheat_rhizo_dist_paov, caption="Bacteria: rhizo wheat")

### soil
bDAT_rare_wheat_soil_dist_paov <- adonis2(bDAT_rare_wheat_soil_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                            data=bDESIGN_wheat_soil, permutations=999)
rownames(bDAT_rare_wheat_soil_dist_paov)[2] <- "Soil chemistry PC1"
rownames(bDAT_rare_wheat_soil_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(bDAT_rare_wheat_soil_dist_paov, caption="Bacteria: soil wheat")

```

**Conclusion:** Soil chemistry PC1 affected the bacterial communities more strongly (sig. in all compartments) compared to BX conditioning. A significant BX effect was found in wheat roots for bacteria. Rhizospere and soil are not affected. 

\pagebreak 

#### Fungi 

```{r PERMANOVA Fungi conditioning, warning=F, echo=F, eval=T, message=F}

### root
fDAT_rare_wheat_root_dist_paov <- adonis2(fDAT_rare_wheat_root_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                          data=fDESIGN_wheat_root, permutations=999)
rownames(fDAT_rare_wheat_root_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_wheat_root_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(fDAT_rare_wheat_root_dist_paov, caption="Fungi: roots wheat")

### rhizo
fDAT_rare_wheat_rhizo_dist_paov <- adonis2(fDAT_rare_wheat_rhizo_dist ~ 
                                             Conditioning * PCA_soil_chem_1, 
                                           data=fDESIGN_wheat_rhizo, permutations=999)
rownames(fDAT_rare_wheat_rhizo_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_wheat_rhizo_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(fDAT_rare_wheat_rhizo_dist_paov, caption="Fungi: rhizo wheat")

### soil
fDAT_rare_wheat_soil_dist_paov <- adonis2(fDAT_rare_wheat_soil_dist ~ 
                                            Conditioning * PCA_soil_chem_1, 
                                          data=fDESIGN_wheat_soil, permutations=999)
rownames(fDAT_rare_wheat_soil_dist_paov)[2] <- "Soil chemistry PC1"
rownames(fDAT_rare_wheat_soil_dist_paov)[3] <- "Genotype : Soil chemistry PC1"  
pander(fDAT_rare_wheat_soil_dist_paov, caption="Fungi: soil wheat")

```

**Conclusion:** Soil chemistry PC1 affected the fungal communities less than the bacteria (lower R2 and only in root and rhizo). The BX conditioning however was more pronounced, being significant in roots and rhizosphere, but not soil samples.


\pagebreak 

## CAP by sampletype  

*Model: ~ Conditioning \* Soil chemistry PC1* 

```{r subsetting phyloseq objects, echo=F, warning=F, message=F, eval=T}

#Bacteria
#bPHYSEQ_rare_wheat
bPHYSEQ_rare_wheat_root <- prune_samples(sample_data(bPHYSEQ_rare_wheat)$Sample_type=="root", bPHYSEQ_rare_wheat)
bPHYSEQ_rare_wheat_rhizo <- prune_samples(sample_data(bPHYSEQ_rare_wheat)$Sample_type=="rhizosphere", bPHYSEQ_rare_wheat)
bPHYSEQ_rare_wheat_soil <- prune_samples(sample_data(bPHYSEQ_rare_wheat)$Sample_type=="soil", bPHYSEQ_rare_wheat)

#Fungi
#fPHYSEQ_rare_wheat
fPHYSEQ_rare_wheat_root <- prune_samples(sample_data(fPHYSEQ_rare_wheat)$Sample_type=="root", fPHYSEQ_rare_wheat)
fPHYSEQ_rare_wheat_rhizo <- prune_samples(sample_data(fPHYSEQ_rare_wheat)$Sample_type=="rhizosphere", fPHYSEQ_rare_wheat)
fPHYSEQ_rare_wheat_soil <- prune_samples(sample_data(fPHYSEQ_rare_wheat)$Sample_type=="soil", fPHYSEQ_rare_wheat)

```

```{r CAPs, echo=F, warning=F, message=F, eval=T}

#function plot CAP
CAP_plot <- function(bPHY_object, CAP_object, title="", stats=""){
  p0 <- phyloseq::plot_ordination(bPHY_object, CAP_object, color='Conditioning', axes = c(1,2), title = title)
  p0$layers <- p0$layers[-1]
  p0 <- p0 + geom_point(aes(size = PCA_soil_chem_1 ), stroke=1, alpha=3/4) 
  p0 <- p0 + stat_ellipse(geom = "polygon", type="t", alpha=0.2, aes(fill=Conditioning))
  p0 <- p0 + scale_fill_manual(labels = c(WT = "WT", bx1 = "bx1"), values=level_cols_Conditioning)
  p0 <- p0 + scale_color_manual(labels = c(WT = "WT", bx1 = "bx1"), values=level_cols_Conditioning)
  #p0 <- p0 + geom_text(label = bDESIGN_wheat_root$PCA_soil_chem_1, size = 3, vjust=2)
  p0 <- p0 + theme_bw(base_size = 12)
  p0 <- p0 + theme(legend.position="none")
  p0 <- p0 + scale_x_reverse()
  p0 <- p0 + labs(subtitle = stats)
  return(p0)
}

#wheat root bacteria
bPHYSEQ_CAP_bray_wheat_root <- phyloseq::ordinate(bPHYSEQ_rare_wheat_root, method="CAP", distance="bray",  
                                                   formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_wheat_root) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_wheat_root, permutations=999) #Anova
bPHYSEQ_CAP_bray_wheat_root_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmroot=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_root_wheat <- CAP_plot(bPHYSEQ_rare_wheat_root, 
                            bPHYSEQ_CAP_bray_wheat_root, "wheat root bacteria", 
                            bPHYSEQ_CAP_bray_wheat_root_stats)


#wheat rhizo bacteria
bPHYSEQ_CAP_bray_wheat_rhizo <- phyloseq::ordinate(bPHYSEQ_rare_wheat_rhizo, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_wheat_rhizo) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_wheat_rhizo, permutations=999) #Anova
bPHYSEQ_CAP_bray_wheat_rhizo_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmrhizo=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_rhizo_wheat <- CAP_plot(bPHYSEQ_rare_wheat_rhizo, 
                            bPHYSEQ_CAP_bray_wheat_rhizo, "wheat rhizo bacteria", 
                            bPHYSEQ_CAP_bray_wheat_rhizo_stats)


#wheat soil bacteria
bPHYSEQ_CAP_bray_wheat_soil <- phyloseq::ordinate(bPHYSEQ_rare_wheat_soil, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(bPHYSEQ_CAP_bray_wheat_soil) # extract variation details
CAP_anova <- anova.cca(bPHYSEQ_CAP_bray_wheat_soil, permutations=999) #Anova
bPHYSEQ_CAP_bray_wheat_soil_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmsoil=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
bCAP_soil_wheat <- CAP_plot(bPHYSEQ_rare_wheat_soil, 
                            bPHYSEQ_CAP_bray_wheat_soil, "wheat soil bacteria", 
                            bPHYSEQ_CAP_bray_wheat_soil_stats)




#wheat root fungi
fPHYSEQ_CAP_bray_wheat_root <- phyloseq::ordinate(fPHYSEQ_rare_wheat_root, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
# stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_wheat_root) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_wheat_root, permutations=999) #Anova
fPHYSEQ_CAP_bray_wheat_root_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmroot=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_root_wheat <- CAP_plot(fPHYSEQ_rare_wheat_root, 
                            fPHYSEQ_CAP_bray_wheat_root, "wheat root fungi", 
                            fPHYSEQ_CAP_bray_wheat_root_stats)


#wheat rhizo fungi
fPHYSEQ_CAP_bray_wheat_rhizo <- phyloseq::ordinate(fPHYSEQ_rare_wheat_rhizo, method="CAP", distance="bray",  
                                                   formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_wheat_rhizo) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_wheat_rhizo, permutations=999) #Anova
fPHYSEQ_CAP_bray_wheat_rhizo_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmrhizo=1),
                                            "% of variance, P = ",
                                            format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_rhizo_wheat <- CAP_plot(fPHYSEQ_rare_wheat_rhizo, 
                             fPHYSEQ_CAP_bray_wheat_rhizo, "wheat rhizo fungi", 
                             fPHYSEQ_CAP_bray_wheat_rhizo_stats)


#wheat soil fungi
fPHYSEQ_CAP_bray_wheat_soil <- phyloseq::ordinate(fPHYSEQ_rare_wheat_soil, method="CAP", distance="bray",  
                                                  formula= ~Conditioning * PCA_soil_chem_1 )
#stats
var_tbl <- variability_table(fPHYSEQ_CAP_bray_wheat_soil) # extract variation details
CAP_anova <- anova.cca(fPHYSEQ_CAP_bray_wheat_soil, permutations=999) #Anova
fPHYSEQ_CAP_bray_wheat_soil_stats <- paste("[", format(var_tbl["constrained", "proportion"] * 100, digits=2, nsmsoil=1),
                                           "% of variance, P = ",
                                           format(CAP_anova[1, 4], digits=2), "]", sep = "")
#plot
fCAP_soil_wheat <- CAP_plot(fPHYSEQ_rare_wheat_soil, 
                            fPHYSEQ_CAP_bray_wheat_soil, "wheat soil fungi", 
                            fPHYSEQ_CAP_bray_wheat_soil_stats)

#legend
for_legend <- phyloseq::plot_ordination(fPHYSEQ_rare_wheat_soil, fPHYSEQ_CAP_bray_wheat_soil, color="Conditioning", axes = c(1,2))
for_legend <- for_legend + geom_point(aes(size = PCA_soil_chem_1 ), stroke=1, alpha=3/4)
for_legend <- for_legend + scale_color_manual(labels = c(WT = "WT", bx1 = "bx1"), values=level_cols_Conditioning)
for_legend <- for_legend + guides(color = guide_legend(order = 1),size = guide_legend(order = 2))
   
```

### Figure 16 | CAP of wheat (by sampletype) 

```{r multiplot CAP, echo=F, warning=F, message=F, eval=T, fig.height=14, fig.width=18}

#combine and plot
CAP_wheat_sample_type <- plot_grid(bCAP_root_wheat, bCAP_rhizo_wheat, bCAP_soil_wheat,
                                   fCAP_root_wheat, fCAP_rhizo_wheat, fCAP_soil_wheat,
                                   align='vh', rel_widths=c(1,1,1,1,1,1),
                                   labels=c("A","B","C","D","F","G"), hjust=-1, ncol=3 )

legend <- get_legend(for_legend + theme(legend.position="bottom") + labs(size="Soil chemistry PC1"))
CAP_multi_wheat_sample_type <- plot_grid(CAP_wheat_sample_type, legend, nrow=2, rel_heights = c(1, 0.2))

print(CAP_multi_wheat_sample_type)

```

**Conclusion:** Soil chemistry PC1 affected the bacterial communities more strongly (sig. in all compartments) compared to BX conditioning. No significant BX effect was found for bacteria. Soil chemistry PC1 affected the fungal communities less than the bacteria (lower R2 and only in and rhizosphere). A BX conditioning effect was found for fungal communities in in roots.

\pagebreak


